app/src/main/java/com/github/se/eduverse/model/MediaCacheManager.kt

```diff
@@ -49,42 +81,53 @@ class MediaCacheManager(
 ) {
   companion object {
     private const val CACHE_LIFETIME_MS = 2 * 24 * 60 * 60 * 1000L // 2 jours
+    // private const val CACHE_LIFETIME_MS = 0L
   }
 
-  fun saveFileToCache(url: String, fileName: String): File? {
-    return try {
-      val file = File(context.cacheDir, fileName)
-      if (!file.exists()) {
-        val inputStream: InputStream = downloader.openStream(url)
-        val outputStream = FileOutputStream(file)
-        inputStream.use { it.copyTo(outputStream) }
-        outputStream.close()
+  @OptIn(UnstableApi::class)
+  suspend fun saveFileToCache(url: String, fileName: String): File? =
+      withContext(Dispatchers.IO) {
+        Log.d("CACHE_DEBUG", "Attempting to download URL: $url")
+        try {
+          if (url.isEmpty()) throw IOException("Empty URL")
+
+          val timestamp = System.currentTimeMillis()
+          val timestampedFileName = "${timestamp}_$fileName"
+
+          val file = File(context.cacheDir, timestampedFileName)
+          val inputStream: InputStream = downloader.openStream(url)
+          val outputStream = FileOutputStream(file)
+
+          inputStream.use { it.copyTo(outputStream) }
+          outputStream.close()
+          Log.d("CACHE_DEBUG", "File saved: $timestampedFileName at ${file.absolutePath}")
+          file
+        } catch (e: Exception) {
+          Log.e("CACHE_ERROR", "Error saving file: ${e.message}")
+          null
+        }
```

Comment:
Provides detailed logging for key actions (download start, file saved, and errors). This helps with debugging and tracing.
Uses inputStream.use to ensure the input stream is closed properly, reducing the risk of resource leaks.
Appends a timestamp to the file name, ensuring uniqueness and avoiding overwriting existing files.
Catches exceptions to prevent crashes and logs the errors for debugging.